src/storage/patch/RepresentationPatchHandler.ts

```diff
@@ -37,10 +46,65 @@ export class RepresentationPatchHandler extends PatchHandler {
       }
       this.logger.debug(`Patching new resource ${identifier.path}`);
     }
-
+    // Converter of RDF text to N3 store
+    const converter = new RdfToQuadConverter();
+    let clonedRepresentation;
+    if (representation) {
+      clonedRepresentation = await cloneRepresentation(representation);
+    }
     // Patch it
     const patched = await this.patcher.handleSafe({ patch, identifier, representation });
 
+    // Solid, §5.3: "Servers MUST NOT allow HTTP POST, PUT and PATCH to update a container’s resource metadata
+    // statements; if the server receives such a request, it MUST respond with a 409 status code.
+    // https://solid.github.io/specification/protocol#contained-resource-metadata-statements
+    if (isContainerIdentifier(identifier)) {
+      throw new ConflictHttpError('Not allowed to execute PATCH request on containers.');
+    }
+
+    if (identifier.path.endsWith('.meta')) {
+      // Convert RDF representation to N3
+      const patchedClone = await cloneRepresentation(patched);
+      const patchedRepresentationQuads = await converter.handleSafe({
+        identifier,
+        representation: patchedClone,
+        preferences: { type: { [INTERNAL_QUADS]: 1 }},
+      });
+      const patchedStore = await readableToQuads(patchedRepresentationQuads.data);
+
+      if (clonedRepresentation) {
+        const representationQuads = await converter.handleSafe({
+          identifier,
+          representation: clonedRepresentation,
+          preferences: { type: { [INTERNAL_QUADS]: 1 }},
+        });
+        const representationStore = await readableToQuads(representationQuads.data);
+        // Test whether pim has been altered
+        const pimStorageIds = representationStore.getSubjects(RDF.type, PIM.Storage, null)
+          .map((subject: Term): string => subject.id);
+        const pimStoragePatchedIds = patchedStore.getSubjects(RDF.type, PIM.Storage, null)
+          .map((subject: Term): string => subject.id);
+
+        let pimStorageNotChanged = true;
```

Comment:
This should not be hardcoded here. There should be an input parameter that contains descriptions of some sort that indicates which triples can not be changed. It could for example take triple patterns as input, and one of the inputs would then be `?x rdf:type pim:storage` (or perhaps a key/value object with optional keys subject/predicate/object to make it more easy to parse). To follow the spec completely it should actually contain the following patterns: `?x rdf:type ?y`, `?x stat:size ?y`, `?x dcterms:modified ?y` and `?x stat:mtime ?y`.